keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java

```diff
@@ -0,0 +1,35 @@
+package io.improbable.keanu.vertices;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+public class LogProbGraph {
+
+    private final Map<Vertex<?>, Vertex<?>> inputs;
+    private final DoubleVertex logProbOutput;
+
+    public LogProbGraph(DoubleVertex logProbOutput) {
+        this.inputs = new HashMap<>();
+        this.logProbOutput = logProbOutput;
+    }
+
+    public <T> LogProbGraph addInput(Vertex<T> from, Vertex<T> to) {
```

Comment:
It would be nice if `LogProbGraph` was immutable. You could create a builder class with an `addInput` method.

Reply:
👍